File: dota-zone/dotazone/src/main/java/br/com/dotazone/view/activity/BuildHeroActivity.java
package br.com.dotazone.view.activity;

import android.app.Dialog;
import android.content.Intent;
import android.content.SharedPreferences;
import android.graphics.Typeface;
import android.os.Bundle;

import androidx.viewpager.widget.ViewPager;
import androidx.drawerlayout.widget.DrawerLayout;

import android.util.Log;
import android.view.View;
import android.view.View.OnClickListener;
import android.widget.Button;
import android.widget.ImageView;
import android.widget.RelativeLayout;
import android.widget.TextView;
import android.widget.Toast;

import com.google.android.gms.analytics.GoogleAnalytics;
import com.viewpagerindicator.PageIndicator;

import br.com.dotazone.BuildConfig;
import br.com.dotazone.DotaZoneApplication;
import br.com.dotazone.DotaZoneBrain;
import br.com.dotazone.DotazoneMenu;
import br.com.dotazone.R;
import br.com.dotazone.model.entity.Hero;
import br.com.dotazone.model.entity.Item;
import br.com.dotazone.model.entity.ItemAtrrib;
import br.com.dotazone.model.listeners.BuildHeroAction;
import br.com.dotazone.model.util.UrlUtils;
import br.com.dotazone.model.util.UtilActivity;
import br.com.dotazone.view.adapter.ItemAdapter;
import br.com.dotazone.view.components.RoundedImageView;

public class BuildHeroActivity extends BaseActivity implements OnClickListener, BuildHeroAction{

	public DotazoneMenu mMenu;
	private ImageView mSlot1;
	private ImageView mSlot2;
	private ImageView mSlot3;
	private ImageView mSlot4;
	private ImageView mSlot5;
	private ImageView mSlot6;
	private ImageView mBuildHelper;
	private boolean mBusySlot1;
	private boolean mBusySlot2;
	private boolean mBusySlot3;
	private boolean mBusySlot4;
	private boolean mBusySlot5;
	private boolean mBusySlot6;
	private Item mItem1;
	private Item mItem2;
	private Item mItem3;
	private Item mItem4;
	private Item mItem5;
	private Item mItem6;
	private TextView mTextInt;
	private TextView mTextAgi;
	private TextView mTextStr;
	private TextView mTextAtk;
	private TextView mTextMove;
	private TextView mTextArmor;
	private TextView mTextIncInt;
	private TextView mTextIncAgi;
	private TextView mTextIncStr;
	private TextView mTextIncAtk;
	private TextView mTextIncMove;
	private TextView mTextIncArmor;
	private TextView mTextHeroName;
	private Hero mHero = DotaZoneBrain.hero;
	private RoundedImageView mHeroIcon;
	private DrawerLayout mDrawerLayout;
	private RelativeLayout mDrawerList;
	private Item mMainBoot;
	private boolean mIsBoot;


	@Override
	protected void onCreate(Bundle savedInstanceState) {
		super.onCreate(savedInstanceState);

		setContentView(R.layout.build_hero_view);
		initComponents();

		mTextAgi.setText(mHero.getAbilites().getAgi()[0]);
		mTextInt.setText(mHero.getAbilites().getInte()[0]);
		mTextStr.setText(mHero.getAbilites().getStr()[0]);

		mTextAtk.setText(mHero.getAbilites().getDmg()[0]);
		mTextMove.setText(mHero.getAbilites().getMs());
		mTextArmor.setText(mHero.getAbilites().getArmor());
		int idImage = getResources().getIdentifier(mHero.getIdString() + "_avatar", "drawable", getPackageName());
		mHeroIcon.setImageResource(idImage);
		Toast.makeText(this, getString(R.string.error_build_alert), Toast.LENGTH_LONG).show();
		((DotaZoneApplication) getApplication()).getTracker(DotaZoneApplication.TrackerName.APP_TRACKER);


	}

	@Override
	protected void onStart() {
		super.onStart();

		GoogleAnalytics.getInstance(this).reportActivityStart(this);
	}

	@Override
	protected void onStop() {
		super.onStop();

		// Stop the analytics tracking
		GoogleAnalytics.getInstance(this).reportActivityStop(this);
	}

	@Override
	protected void onActivityResult(int requestCode, int resultCode, Intent data) {
		super.onActivityResult(requestCode, resultCode, data);
		// Pass on the activity result to the helper for handling
	}

	@Override
	protected void onResume() {
		super.onResume();

		final SharedPreferences settings = getSharedPreferences(UrlUtils.PREFS_NAME, 0);
		boolean isHelp = settings.getBoolean("help_build_hero", true);

		if (isHelp) {

			UtilActivity.showHelpBuildHero(this);
			SharedPreferences.Editor editor = settings.edit();
			editor.putBoolean("help_build_hero", false);
			editor.commit();

		}


	}

	@Override
	public void initComponents() {


		mSlot1 = (ImageView) findViewById(R.id.build_hero_slot1);
		mSlot1.setOnClickListener(this);

		mSlot2 = (ImageView) findViewById(R.id.build_hero_slot2);
		mSlot2.setOnClickListener(this);

		mSlot3 = (ImageView) findViewById(R.id.build_hero_slot3);
		mSlot3.setOnClickListener(this);

		mSlot4 = (ImageView) findViewById(R.id.build_hero_slot4);
		mSlot4.setOnClickListener(this);

		mSlot5 = (ImageView) findViewById(R.id.build_hero_slot5);
		mSlot5.setOnClickListener(this);

		mSlot6 = (ImageView) findViewById(R.id.build_hero_slot6);
		mSlot6.setOnClickListener(this);

		mBuildHelper = (ImageView) findViewById(R.id.build_hero_help);
		mBuildHelper.setOnClickListener(new OnClickListener() {
			@Override
			public void onClick(View view) {
				UtilActivity.showHelpBuildHero(BuildHeroActivity.this);
			}
		});

		mTextAgi = (TextView) findViewById(R.id.hero_text_attribute_agi);
		mTextStr = (TextView) findViewById(R.id.hero_text_attribute_str);
		mTextInt = (TextView) findViewById(R.id.hero_text_attribute_int);
		mTextAtk = (TextView) findViewById(R.id.hero_text_attribute_atk);
		mTextMove = (TextView) findViewById(R.id.hero_text_attribute_move);
		mTextArmor = (TextView) findViewById(R.id.hero_text_attribute_armor);
		mTextHeroName = (TextView) findViewById(R.id.hero_text_name);

		mHeroIcon = (RoundedImageView) findViewById(R.id.build_hero_icon);

		mTextIncAgi = (TextView) findViewById(R.id.hero_text_attribute_add_agi);
		mTextIncStr = (TextView) findViewById(R.id.hero_text_attribute_add_str);
		mTextIncInt = (TextView) findViewById(R.id.hero_text_attribute_add_int);
		mTextIncAtk = (TextView) findViewById(R.id.hero_text_attribute_add_atk);
		mTextIncMove = (TextView) findViewById(R.id.hero_text_attribute_add_move);
		mTextIncArmor = (TextView) findViewById(R.id.hero_text_attribute_add_armor);

		ViewPager viewPager = (ViewPager) findViewById(R.id.build_item_pager);
		PageIndicator indicator = (PageIndicator) findViewById(R.id.indicator_build_hero);
		ItemAdapter itemAdapter = new ItemAdapter(getSupportFragmentManager(), this);
		viewPager.setAdapter(itemAdapter);
		indicator.setViewPager(viewPager);

		String nameHero = mHero.getName().replace("_", " ");
		mTextHeroName.setText(nameHero);
		Typeface font = Typeface.createFromAsset(getAssets(), "Roboto-Thin.ttf");
		mTextHeroName.setTypeface(font);

		mDrawerLayout = (DrawerLayout) findViewById(R.id.drawer_layout);
		mDrawerList = (RelativeLayout) findViewById(R.id.list_slidermenu);
		mMenu = new DotazoneMenu(this, mDrawerLayout, mDrawerList);
		mMenu.checkBuildMenu();
	}

	@Override
	public void setActionErrorOk() {
		// TODO Auto-generated method stub

	}

	@Override
	public void setActionErrorCancel() {
		// TODO Auto-generated method stub

	}


	@Override
	public void onClick(View v) {

		//TODO remover antes de subir
		if (/*DotaZoneBrain.isPremium*/true) {

			switch (v.getId()) {


				case R.id.build_hero_slot1:

					mSlot1.setBackgroundResource(R.drawable.empty);
					mBusySlot1 = false;
					if (mItem1 != null)
						decrementValuesInAttribute(mItem1.getItemAttrib());
					mItem1 = null;

					break;
				case R.id.build_hero_slot2:

					mSlot2.setBackgroundResource(R.drawable.empty);
					mBusySlot2 = false;
					if (mItem2 != null)
						decrementValuesInAttribute(mItem2.getItemAttrib());
					mItem2 = null;
					break;

				case R.id.build_hero_slot3:

					mSlot3.setBackgroundResource(R.drawable.empty);
					mBusySlot3 = false;
					if (mItem3 != null)
						decrementValuesInAttribute(mItem3.getItemAttrib());
					mItem3 = null;
					break;

				case R.id.build_hero_slot4:

					mSlot4.setBackgroundResource(R.drawable.empty);
					mBusySlot4 = false;
					if (mItem4 != null)
						decrementValuesInAttribute(mItem4.getItemAttrib());
					mItem4 = null;
					break;

				case R.id.build_hero_slot5:

					mSlot5.setBackgroundResource(R.drawable.empty);
					mBusySlot5 = false;
					if (mItem5 != null)
						decrementValuesInAttribute(mItem5.getItemAttrib());
					mItem5 = null;
					break;

				case R.id.build_hero_slot6:

					mSlot6.setBackgroundResource(R.drawable.empty);
					mBusySlot6 = false;
					if (mItem6 != null)
						decrementValuesInAttribute(mItem6.getItemAttrib());
					mItem6 = null;
					break;

				default:
					break;
			}
		} else {


		}


	}

	@Override
	public void addingItemSlot(Item item) {

		int pos = item.getImageName().lastIndexOf('.');
		int idImage = getResources().getIdentifier(item.getImageName().substring(0, pos), "drawable", getPackageName());

		if (DotaZoneBrain.isPremium) {

			if (!mBusySlot1 && item != null) {
				mSlot1.setBackgroundResource(idImage);
				incrementValuesInAttribute(item);
				mItem1 = item;
				mBusySlot1 = true;
				mSlot1.setOnLongClickListener(new View.OnLongClickListener() {
					@Override
					public boolean onLongClick(View view) {
						if (mItem1 != null)
							UtilActivity.showDescriptionItem(mItem1, BuildHeroActivity.this);
						return false;
					}
				});

			} else if (!mBusySlot2 && item != null) {
				mSlot2.setBackgroundResource(idImage);
				mBusySlot2 = true;
				mItem2 = item;
				incrementValuesInAttribute(item);
				mSlot2.setOnLongClickListener(new View.OnLongClickListener() {
					@Override
					public boolean onLongClick(View view) {
						if (mItem2 != null)
							UtilActivity.showDescriptionItem(mItem2, BuildHeroActivity.this);
						return false;
					}
				});

			} else if (!mBusySlot3 && item != null) {
				mSlot3.setBackgroundResource(idImage);
				mBusySlot3 = true;
				mItem3 = item;
				incrementValuesInAttribute(item);
				mSlot3.setOnLongClickListener(new View.OnLongClickListener() {
					@Override
					public boolean onLongClick(View view) {
						if (mItem3 != null)
							UtilActivity.showDescriptionItem(mItem3, BuildHeroActivity.this);
						return false;
					}
				});

			} else if (!mBusySlot4 && item != null) {
				mSlot4.setBackgroundResource(idImage);
				mBusySlot4 = true;
				mItem4 = item;
				incrementValuesInAttribute(item);
				mSlot4.setOnLongClickListener(new View.OnLongClickListener() {
					@Override
					public boolean onLongClick(View view) {
						if (mItem4 != null)
							UtilActivity.showDescriptionItem(mItem4, BuildHeroActivity.this);
						return false;

					}
				});

			} else if (!mBusySlot5 && item != null) {
				mSlot5.setBackgroundResource(idImage);
				mBusySlot5 = true;
				mItem5 = item;
				incrementValuesInAttribute(item);
				mSlot5.setOnLongClickListener(new View.OnLongClickListener() {
					@Override
					public boolean onLongClick(View view) {
						if (mItem5 != null)
							UtilActivity.showDescriptionItem(mItem5, BuildHeroActivity.this);

						return false;
					}
				});

			} else if (!mBusySlot6 && item != null) {
				mSlot6.setBackgroundResource(idImage);
				mBusySlot6 = true;
				mItem6 = item;
				incrementValuesInAttribute(item);
				mSlot6.setOnLongClickListener(new View.OnLongClickListener() {
					@Override
					public boolean onLongClick(View view) {
						if (mItem6 != null)
							UtilActivity.showDescriptionItem(mItem6, BuildHeroActivity.this);

						return false;
					}
				});

			} else {

				Toast.makeText(this, getString(R.string.full_slot), Toast.LENGTH_SHORT).show();
			}
		} else {
			showAlertBuyProVersion();
		}


	}

	@Override
	public void verifyPayment() {
	}

	private void decrementValuesInAttribute(ItemAtrrib item) {

		if (item != null) {

			int valueInt = mTextIncInt.getText().equals("") ? 0 : Integer.valueOf(UrlUtils.extractNumber(mTextIncInt.getText().toString()));
			int valueAgi = mTextIncAgi.getText().equals("") ? 0 : Integer.valueOf(UrlUtils.extractNumber(mTextIncAgi.getText().toString()));
			int valueStr = mTextIncStr.getText().equals("") ? 0 : Integer.valueOf(UrlUtils.extractNumber(mTextIncStr.getText().toString()));
			int valueAtk = mTextIncAtk.getText().equals("") ? 0 : Integer.valueOf(UrlUtils.extractNumber(mTextIncAtk.getText().toString()));
			int valueMs = mTextIncMove.getText().equals("") ? 0 : Integer.valueOf(UrlUtils.extractNumber(mTextIncMove.getText().toString()));
			int valueArmor = mTextIncArmor.getText().equals("") ? 0 : Integer.valueOf(UrlUtils.extractNumber(mTextIncArmor.getText().toString()));

			int additionalInt = item.getInteligence();
			int additionalAgi = item.getAgility();
			int additionalStr = item.getStrength();
			int additionalAtk = item.getDamage();
			int additionalMs = Integer.valueOf(item.getMs());
			int additionalArmor = item.getArmor();

			mTextIncAgi.setText(Integer.valueOf(Math.abs(valueAgi - additionalAgi)) == 0 ? String.valueOf("") : " + "
					+ Integer.valueOf(Math.abs(valueAgi - additionalAgi)).toString());
			mTextIncInt.setText(Integer.valueOf(Math.abs(valueInt - additionalInt)) == 0 ? String.valueOf("") : " + "
					+ Integer.valueOf(Math.abs(valueInt - additionalInt)).toString());
			mTextIncStr.setText(Integer.valueOf(Math.abs(valueStr - additionalStr)) == 0 ? String.valueOf("") : " + "
					+ Integer.valueOf(Math.abs(valueStr - additionalStr)).toString());

			if (mMainBoot.getItemAttrib().getId().equals(item.getId()) && additionalMs != 0 && mIsBoot) {
				mTextIncMove.setText(Integer.valueOf(Math.abs(valueMs - additionalMs)) == 0 ? String.valueOf("") : " + "
						+ Integer.valueOf(Math.abs(valueMs - additionalMs)).toString());
				mIsBoot = false;
			}

			mTextIncArmor.setText(Integer.valueOf(Math.abs(valueArmor - additionalArmor)) == 0 ? String.valueOf("") : " + "
					+ Integer.valueOf(Math.abs(valueArmor - additionalArmor)).toString());

			mTextIncAtk.setText(Integer.valueOf(Math.abs(additionalAtk - valueAtk)) == 0 ? String.valueOf("") : " + "
					+ Integer.valueOf(Math.abs(valueAtk - additionalAtk)).toString());
		}
	}

	private void incrementValuesInAttribute(Item item) {

		if (item != null && item.getItemAttrib() != null) {

			int valueInt = mTextIncInt.getText().equals("") ? 0 : Integer.valueOf(UrlUtils.extractNumber(mTextIncInt.getText().toString()));
			int valueAgi = mTextIncAgi.getText().equals("") ? 0 : Integer.valueOf(UrlUtils.extractNumber(mTextIncAgi.getText().toString()));
			int valueStr = mTextIncStr.getText().equals("") ? 0 : Integer.valueOf(UrlUtils.extractNumber(mTextIncStr.getText().toString()));
			int valueAtk = mTextIncAtk.getText().equals("") ? 0 : Integer.valueOf(UrlUtils.extractNumber(mTextIncAtk.getText().toString()));
			int valueMs = mTextIncMove.getText().equals("") ? 0 : Integer.valueOf(UrlUtils.extractNumber(mTextIncMove.getText().toString()));
			int valueArmor = mTextIncArmor.getText().equals("") ? 0 : Integer.valueOf(UrlUtils.extractNumber(mTextIncArmor.getText().toString()));

			int additionalInt = item.getItemAttrib().getInteligence();
			int additionalAgi = item.getItemAttrib().getAgility();
			int additionalStr = item.getItemAttrib().getStrength();
			int additionalAtk = item.getItemAttrib().getDamage();
			int additionalMs = Integer.valueOf(item.getItemAttrib().getMs());
			int additionalArmor = item.getItemAttrib().getArmor();

			mTextIncAgi.setText(Integer.valueOf(valueAgi + additionalAgi) == 0 ? String.valueOf("") : " + "
					+ Integer.valueOf(valueAgi + additionalAgi).toString());
			mTextIncInt.setText(Integer.valueOf(valueInt + additionalInt) == 0 ? String.valueOf("") : " + "
					+ Integer.valueOf(valueInt + additionalInt).toString());
			mTextIncStr.setText(Integer.valueOf(valueStr + additionalStr) == 0 ? String.valueOf("") : " + "
					+ Integer.valueOf(valueStr + additionalStr).toString());

			mTextIncAtk.setText(Integer.valueOf(valueAtk + additionalAtk) == 0 ? String.valueOf("") : " + "
					+ Integer.valueOf(valueAtk + additionalAtk).toString());

			if (valueMs == 0) {

				mTextIncMove.setText(Integer.valueOf(valueMs + additionalMs) == 0 ? String.valueOf("") : " + "
						+ Integer.valueOf(valueMs + additionalMs).toString());
				mMainBoot = item;
				mIsBoot = true;
			}

			mTextIncArmor.setText(Integer.valueOf(valueArmor + additionalArmor) == 0 ? String.valueOf("") : " + "
					+ Integer.valueOf(valueArmor + additionalArmor).toString());

		}
	}

	private void showAlertBuyProVersion() {


		final Dialog dialog = new Dialog(this, R.style.FullHeightDialog);
		dialog.show();
		dialog.setContentView(R.layout.alert_buy_view);
		dialog.setCancelable(false);
		Button buy = (Button) dialog.findViewById(R.id.buy_pro_version);
		Button close = (Button) dialog.findViewById(R.id.buy_close);

		close.setOnClickListener(new OnClickListener() {

			@Override
			public void onClick(View v) {

				dialog.dismiss();
			}
		});
		buy.setOnClickListener(new OnClickListener() {

			@Override
			public void onClick(View v) {
				dialog.dismiss();
			}
		});
	}
}

File: dota-zone/app/src/main/java/com/br/dotazone/view/activity/BuildHeroActivity.kt
package com.br.dotazone.view.activity

import android.app.Dialog
import android.graphics.Typeface
import android.os.Bundle
import android.view.View
import android.widget.*
import androidx.drawerlayout.widget.DrawerLayout
import androidx.viewpager.widget.ViewPager
import com.br.dotazone.DotaZoneApplication
import com.br.dotazone.DotaZoneBrain.hero
import com.br.dotazone.DotaZoneBrain.isPremium
import com.br.dotazone.R
import com.br.dotazone.model.entity.Item
import com.br.dotazone.model.entity.ItemAtrrib
import com.br.dotazone.model.listeners.BuildHeroAction
import com.br.dotazone.model.util.UrlUtils
import com.br.dotazone.model.util.UtilActivity
import com.br.dotazone.view.adapter.ItemAdapter
import com.com.dotazone.DotazoneMenu
import com.google.android.gms.analytics.GoogleAnalytics
import com.google.android.material.tabs.TabLayout
import kotlinx.android.synthetic.main.build_hero_view.*
import kotlinx.android.synthetic.main.hero_attributes_build.*
import kotlinx.android.synthetic.main.hero_profile_view.*
import kotlin.math.abs


class BuildHeroActivity : BaseActivity(), BuildHeroAction {

	var mMenu: DotazoneMenu? = null
	private var mBusySlot1 = false
	private var mBusySlot2 = false
	private var mBusySlot3 = false
	private var mBusySlot4 = false
	private var mBusySlot5 = false
	private var mBusySlot6 = false
	private var mItem1: Item? = null
	private var mItem2: Item? = null
	private var mItem3: Item? = null
	private var mItem4: Item? = null
	private var mItem5: Item? = null
	private var mItem6: Item? = null
	private val mHero = hero
	private var mDrawerLayout: DrawerLayout? = null
	private var mDrawerList: RelativeLayout? = null
	private var mMainBoot: Item? = null
	private var mIsBoot = false

	override fun onCreate(savedInstanceState: Bundle?) {
		super.onCreate(savedInstanceState)
		setContentView(R.layout.build_hero_view)
		initComponents()
		heroTextAttributeAgi.text = mHero?.abilites?.agi?.get(0) ?:""
		heroTextAttributeInt.text = mHero?.abilites?.inte?.get(0) ?: ""
		heroTextAttributeStr.text = mHero?.abilites?.str?.get(0) ?: ""
		heroTextAttributeAtk.text = mHero?.abilites?.dmg?.get(0) ?: ""
		heroTextAttributeMove.text = mHero?.abilites?.ms ?: ""
		heroTextAttributeArmor.text = mHero?.abilites?.armor ?:""
		val idImage = resources.getIdentifier((mHero?.idString ?: "") + "_avatar", "drawable", packageName)
		buildHeroCircleImage.setImageResource(idImage)
		Toast.makeText(this, getString(R.string.error_build_alert), Toast.LENGTH_LONG).show()
	}

	override fun onStart() {
		super.onStart()
		GoogleAnalytics.getInstance(this).reportActivityStart(this)
	}

	override fun onStop() {
		super.onStop()
		// Stop the analytics tracking
		GoogleAnalytics.getInstance(this).reportActivityStop(this)
	}
	override fun onResume() {
		super.onResume()
		val settings = getSharedPreferences(UrlUtils.PREFS_NAME, 0)
		val isHelp = settings.getBoolean("help_build_hero", true)
		if (isHelp) {
			UtilActivity.showHelpBuildHero(this)
			val editor = settings.edit()
			editor.putBoolean("help_build_hero", false)
			editor.commit()
		}
	}

	override fun initComponents() {

		buildHeroSlot1.setOnClickListener {
			buildHeroSlot1.setBackgroundResource(R.drawable.empty)
			mBusySlot1 = false
			if (mItem1 != null) decrementValuesInAttribute(mItem1!!.itemAttrib)
			mItem1 = null
		}
		buildHeroSlot2.setOnClickListener {
			buildHeroSlot2.setBackgroundResource(R.drawable.empty)
			mBusySlot2 = false
			if (mItem2 != null) decrementValuesInAttribute(mItem2!!.itemAttrib)
			mItem2 = null
		}
		buildHeroSlot3.setOnClickListener {
			buildHeroSlot3.setBackgroundResource(R.drawable.empty)
			mBusySlot3 = false
			if (mItem3 != null) decrementValuesInAttribute(mItem3!!.itemAttrib)
			mItem3 = null
		}
		buildHeroSlot4.setOnClickListener {
			buildHeroSlot4.setBackgroundResource(R.drawable.empty)
			mBusySlot4 = false
			if (mItem4 != null) decrementValuesInAttribute(mItem4!!.itemAttrib)
			mItem4 = null
		}
		buildHeroSlot5.setOnClickListener {
			buildHeroSlot5.setBackgroundResource(R.drawable.empty)
			mBusySlot5 = false
			if (mItem5 != null) decrementValuesInAttribute(mItem5!!.itemAttrib)
			mItem5 = null
		}
		buildHeroSlot6.setOnClickListener {
			buildHeroSlot6.setBackgroundResource(R.drawable.empty)
			mBusySlot6 = false
			if (mItem6 != null) decrementValuesInAttribute(mItem6!!.itemAttrib)
			mItem6 = null
		}
		buildHeroHelp.setOnClickListener { UtilActivity.showHelpBuildHero(this@BuildHeroActivity) }
		val viewPager = findViewById<View>(R.id.buildItemPager) as ViewPager
		val indicator = findViewById<View>(R.id.indicatorBuildHero) as TabLayout
		val itemAdapter = ItemAdapter(supportFragmentManager, this)
		viewPager.adapter = itemAdapter
		indicator.setupWithViewPager(viewPager)
		val nameHero = mHero!!.name?.replace("_", " ")
		heroTextNameView.text = nameHero
		val font = Typeface.createFromAsset(assets, "Roboto-Thin.ttf")
		heroTextAttributeAddStr.typeface = font
		mDrawerLayout = findViewById<View>(R.id.drawer_layout) as DrawerLayout
		mDrawerList = findViewById<View>(R.id.listSliderMenu) as RelativeLayout
		mMenu = DotazoneMenu(this, mDrawerLayout, mDrawerList)
		mMenu!!.checkBuildMenu()
	}

	override fun setActionErrorOk() {
		// TODO Auto-generated method stub
	}

	override fun setActionErrorCancel() {
		// TODO Auto-generated method stub
	}

	override fun addingItemSlot(item: Item?) {
		val pos = item?.imageName?.lastIndexOf('.')?:0
		val idImage = resources.getIdentifier(item?.imageName?.substring(0, pos), "drawable", packageName)
		if (isPremium) {
			if (!mBusySlot1 && item != null) {
				buildHeroSlot1.setBackgroundResource(idImage)
				incrementValuesInAttribute(item)
				mItem1 = item
				mBusySlot1 = true
				buildHeroSlot1.setOnLongClickListener {
					if (mItem1 != null) UtilActivity.showDescriptionItem(mItem1, this@BuildHeroActivity)
					false
				}
			} else if (!mBusySlot2 && item != null) {
				buildHeroSlot2.setBackgroundResource(idImage)
				mBusySlot2 = true
				mItem2 = item
				incrementValuesInAttribute(item)
				buildHeroSlot2.setOnLongClickListener {
					if (mItem2 != null) UtilActivity.showDescriptionItem(mItem2, this@BuildHeroActivity)
					false
				}
			} else if (!mBusySlot3 && item != null) {
				buildHeroSlot3.setBackgroundResource(idImage)
				mBusySlot3 = true
				mItem3 = item
				incrementValuesInAttribute(item)
				buildHeroSlot3.setOnLongClickListener {
					if (mItem3 != null) UtilActivity.showDescriptionItem(mItem3, this@BuildHeroActivity)
					false
				}
			} else if (!mBusySlot4 && item != null) {
				buildHeroSlot4.setBackgroundResource(idImage)
				mBusySlot4 = true
				mItem4 = item
				incrementValuesInAttribute(item)
				buildHeroSlot4.setOnLongClickListener {
					if (mItem4 != null) UtilActivity.showDescriptionItem(mItem4, this@BuildHeroActivity)
					false
				}
			} else if (!mBusySlot5 && item != null) {
				buildHeroSlot5.setBackgroundResource(idImage)
				mBusySlot5 = true
				mItem5 = item
				incrementValuesInAttribute(item)
				buildHeroSlot5.setOnLongClickListener {
					if (mItem5 != null) UtilActivity.showDescriptionItem(mItem5, this@BuildHeroActivity)
					false
				}
			} else if (!mBusySlot6 && item != null) {
				buildHeroSlot6.setBackgroundResource(idImage)
				mBusySlot6 = true
				mItem6 = item
				incrementValuesInAttribute(item)
				buildHeroSlot6.setOnLongClickListener {
					if (mItem6 != null) UtilActivity.showDescriptionItem(mItem6, this@BuildHeroActivity)
					false
				}
			} else {
				Toast.makeText(this, getString(R.string.full_slot), Toast.LENGTH_SHORT).show()
			}
		} else {
			showAlertBuyProVersion()
		}
	}

	override fun verifyPayment() {}
	private fun decrementValuesInAttribute(item: ItemAtrrib?) {
		if (item != null) {
			val valueInt = if (heroTextAttributeAddInt.text == "") 0 else Integer.valueOf(UrlUtils.extractNumber(heroTextAttributeAddInt.text.toString()))
			val valueAgi = if (heroTextAttributeAddAgi.text == "") 0 else Integer.valueOf(UrlUtils.extractNumber(heroTextAttributeAddAgi.text.toString()))
			val valueStr = if (heroTextAttributeAddInt.text == "") 0 else Integer.valueOf(UrlUtils.extractNumber(heroTextAttributeAddInt.text.toString()))
			val valueAtk = if (heroTextAttributeAddAtk.text == "") 0 else Integer.valueOf(UrlUtils.extractNumber(heroTextAttributeAddAtk.text.toString()))
			val valueMs = if (heroTextAttributeAddMove.text == "") 0 else Integer.valueOf(UrlUtils.extractNumber(heroTextAttributeAddMove.text.toString()))
			val valueArmor = if (heroTextAttributeAddArmor.text == "") 0 else Integer.valueOf(UrlUtils.extractNumber(heroTextAttributeAddArmor.text.toString()))
			val additionalInt = item.inteligence
			val additionalAgi = item.agility
			val additionalStr = item.strength
			val additionalAtk = item.damage
			val additionalMs = Integer.valueOf(item.ms)
			val additionalArmor = item.armor
			heroTextAttributeAddAgi.text = if (Integer.valueOf(abs(valueAgi - additionalAgi)) == 0) "" else " + "+Integer.valueOf(abs(valueAgi - additionalAgi)).toString()
			heroTextAttributeAddInt.text = if (Integer.valueOf(abs(valueInt - additionalInt)) == 0) "" else " + "+Integer.valueOf(abs(valueInt - additionalInt)).toString()
			heroTextAttributeAddInt.text = if (Integer.valueOf(abs(valueStr - additionalStr)) == 0) "" else " + "+Integer.valueOf(abs(valueStr - additionalStr)).toString()
			if (mMainBoot!!.itemAttrib?.id == item.id && additionalMs != 0 && mIsBoot) {
				heroTextAttributeAddMove.text = if (Integer.valueOf(abs(valueMs - additionalMs)) == 0) "" else " + "+Integer.valueOf(abs(valueMs - additionalMs)).toString()
				mIsBoot = false
			}
			heroTextAttributeAddArmor.text = if (Integer.valueOf(abs(valueArmor - additionalArmor)) == 0) "" else " + "+Integer.valueOf(abs(valueArmor - additionalArmor)).toString()
			heroTextAttributeAddAtk.text = if (Integer.valueOf(abs(additionalAtk - valueAtk)) == 0) "" else " + "+Integer.valueOf(abs(valueAtk - additionalAtk)).toString()
		}
	}

	private fun incrementValuesInAttribute(item: Item?) {
		if (item != null && item.itemAttrib != null) {
			val valueInt = if (heroTextAttributeAddInt.text == "") 0 else Integer.valueOf(UrlUtils.extractNumber(heroTextAttributeAddInt.text.toString()))
			val valueAgi = if (heroTextAttributeAddAgi.text == "") 0 else Integer.valueOf(UrlUtils.extractNumber(heroTextAttributeAddAgi.text.toString()))
			val valueStr = if (heroTextAttributeAddStr.text == "") 0 else Integer.valueOf(UrlUtils.extractNumber(heroTextAttributeAddStr.text.toString()))
			val valueAtk = if (heroTextAttributeAddAtk.text == "") 0 else Integer.valueOf(UrlUtils.extractNumber(heroTextAttributeAddAtk.text.toString()))
			val valueMs = if (heroTextAttributeAddMove.text == "") 0 else Integer.valueOf(UrlUtils.extractNumber(heroTextAttributeAddMove.text.toString()))
			val valueArmor = if (heroTextAttributeAddArmor.text == "") 0 else Integer.valueOf(UrlUtils.extractNumber(heroTextAttributeAddArmor.text.toString()))
			val additionalInt = item.itemAttrib?.inteligence
			val additionalAgi = item.itemAttrib?.agility
			val additionalStr = item.itemAttrib?.strength
			val additionalAtk = item.itemAttrib?.damage
			val additionalMs = Integer.valueOf(item.itemAttrib?.ms?:"0")
			val additionalArmor = item.itemAttrib?.armor
			heroTextAttributeAddAgi.text = if (Integer.valueOf(valueAgi + additionalAgi!!) == 0) "" else " + "+Integer.valueOf(valueAgi + additionalAgi).toString()
			heroTextAttributeAddInt.text = if (Integer.valueOf(valueInt + additionalInt!!) == 0) "" else " + "+Integer.valueOf(valueInt + additionalInt).toString()
			heroTextAttributeAddStr.text = if (Integer.valueOf(valueStr + additionalStr!!) == 0) "" else " + "+Integer.valueOf(valueStr + additionalStr).toString()
			heroTextAttributeAddAtk.text = if (Integer.valueOf(valueAtk + additionalAtk!!) == 0) "" else " + "+Integer.valueOf(valueAtk + additionalAtk).toString()
			if (valueMs == 0) {
				heroTextAttributeAddMove.text = if (Integer.valueOf(valueMs + additionalMs) == 0) "" else " + " + Integer.valueOf(valueMs + additionalMs).toString()
				mMainBoot = item
				mIsBoot = true
			}
			heroTextAttributeAddArmor.text = if (Integer.valueOf(valueArmor + additionalArmor!!) == 0) "" else " + "+Integer.valueOf(valueArmor + additionalArmor).toString()
		}
	}

	private fun showAlertBuyProVersion() {
		val dialog = Dialog(this, R.style.FullHeightDialog)
		dialog.show()
		dialog.setContentView(R.layout.alert_buy_view)
		dialog.setCancelable(false)
		val buy = dialog.findViewById<View>(R.id.buy_pro_version) as Button
		val close = dialog.findViewById<View>(R.id.buy_close) as Button
		close.setOnClickListener { dialog.dismiss() }
		buy.setOnClickListener { dialog.dismiss() }
	}
}

